File: new/atoms/paid-in-range.test.mjs
import { paidInRange } from './paid-in-range.mjs';
let f = 0;
const ok = (cond, msg) => { if (!cond) { console.error('✗ ' + msg); f = 1; } };
// שקע-הטווח כהתנהגות-המקור (inRange של reports/lib)
const ir = (iso, r) => {
  if (!iso) return false;
  if (r.from && iso < r.from) return false;
  if (r.to && iso > r.to) return false;
  return true;
};
const jan = { from: '2026-01-01', to: '2026-01-31' };
// 1) רק תשלום-ינואר
const e1 = { payments: [{ date: '2026-01-10', amount: 100 }, { date: '2026-02-10', amount: 50 }] };
ok(paidInRange(e1, jan, ir) === 100, 'דוגמה 1');
// 2) טווח פתוח ⇒ הכול
ok(paidInRange(e1, {}, ir) === 150, 'דוגמה 2');
// 3) גבולות סגורים משני הקצוות
const e3 = { payments: [{ date: '2026-01-01', amount: 10 }, { date: '2026-01-31', amount: 20 }] };
ok(paidInRange(e3, jan, ir) === 30, 'דוגמה 3');
// 4) בלי payments / ריק ⇒ 0
ok(paidInRange({}, jan, ir) === 0, 'דוגמה 4א: e ריק');
ok(paidInRange({ payments: [] }, jan, ir) === 0, 'דוגמה 4ב: מערך ריק');
// 5) מגן-מספר: NaN ומחרוזת ⇒ 0
const e5 = {
  payments: [
    { date: '2026-01-10', amount: NaN },
    { date: '2026-01-11', amount: '80' },
    { date: '2026-01-12', amount: 25 },
  ],
};
ok(paidInRange(e5, jan, ir) === 25, 'דוגמה 5');
// 6) תשלום בלי תאריך מסונן ע"י השקע
ok(paidInRange({ payments: [{ date: '', amount: 40 }] }, jan, ir) === 0, 'דוגמה 6');
if (f) process.exit(1);
console.log('✓ paid-in-range: 6 דוגמאות-חוזה — ירוק');

File: new/dart-maor/paid-in-range_test.dart
// רתמת-זהב · paid-in-range — assert-ים = דוגמאות-החוזה של בדיקת-ה-JS (זהות).
// אם עובר: Dart≡JS. הקלטים/פלטים הועתקו verbatim מ-new/atoms/paid-in-range.test.mjs.
// הבדיקה מייבאת אך ורק את האטום שלה (חוק-4); שקע inRange משוכפל מבדיקת-ה-JS.
import 'paid-in-range.dart';

// שקע-הטווח כהתנהגות-המקור (inRange של reports/lib), שעתוק-אמת של ה-ir בבדיקת-ה-JS:
//   if (!iso) return false; if (r.from && iso < r.from) return false;
//   if (r.to && iso > r.to) return false; return true;
bool _ir(Object? iso, Map<String, dynamic> r) {
  if (iso == null) return false; // !iso
  final String s = iso as String;
  if (s.isEmpty) return false; // מחרוזת-ריקה = falsy של JS
  final Object? from = r['from'];
  if (from is String && from.isNotEmpty && s.compareTo(from) < 0) return false;
  final Object? to = r['to'];
  if (to is String && to.isNotEmpty && s.compareTo(to) > 0) return false;
  return true;
}

void main() {
  final jan = <String, dynamic>{'from': '2026-01-01', 'to': '2026-01-31'};

  // 1) רק תשלום-ינואר
  final e1 = <String, dynamic>{
    'payments': [
      {'date': '2026-01-10', 'amount': 100},
      {'date': '2026-02-10', 'amount': 50},
    ],
  };
  assert(paidInRange(e1, jan, _ir) == 100, 'דוגמה 1');

  // 2) טווח פתוח ⇒ הכול
  assert(paidInRange(e1, <String, dynamic>{}, _ir) == 150, 'דוגמה 2');

  // 3) גבולות סגורים משני הקצוות
  final e3 = <String, dynamic>{
    'payments': [
      {'date': '2026-01-01', 'amount': 10},
      {'date': '2026-01-31', 'amount': 20},
    ],
  };
  assert(paidInRange(e3, jan, _ir) == 30, 'דוגמה 3');

  // 4) בלי payments / ריק ⇒ 0
  assert(paidInRange(<String, dynamic>{}, jan, _ir) == 0, 'דוגמה 4א: e ריק');
  assert(paidInRange(<String, dynamic>{'payments': []}, jan, _ir) == 0, 'דוגמה 4ב: מערך ריק');

  // 5) מגן-מספר: NaN ומחרוזת ⇒ 0
  final e5 = <String, dynamic>{
    'payments': [
      {'date': '2026-01-10', 'amount': double.nan},
      {'date': '2026-01-11', 'amount': '80'},
      {'date': '2026-01-12', 'amount': 25},
    ],
  };
  assert(paidInRange(e5, jan, _ir) == 25, 'דוגמה 5');

  // 6) תשלום בלי תאריך מסונן ע"י השקע
  assert(
    paidInRange(<String, dynamic>{
      'payments': [
        {'date': '', 'amount': 40},
      ],
    }, jan, _ir) == 0,
    'דוגמה 6',
  );

  print('✓ paid-in-range: 6 דוגמאות-חוזה — Dart≡JS ירוק');
}
